provider/win_driver_installer_test.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_misc
+
+from provider import win_driver_utils
+
+driver_name_list = ['netkvm', 'viorng', 'vioser',
+                    'balloon', 'pvpanic', 'vioinput',
+                    'viofs', 'viostor', 'vioscsi']
+
+device_hwid_list = ['"PCI\\VEN_1AF4&DEV_1000" "PCI\\VEN_1AF4&DEV_1041"',
+                    '"PCI\\VEN_1AF4&DEV_1005" "PCI\\VEN_1AF4&DEV_1044"',
+                    '"PCI\\VEN_1AF4&DEV_1003" "PCI\\VEN_1AF4&DEV_1043"',
+                    '"PCI\\VEN_1AF4&DEV_1002" "PCI\\VEN_1AF4&DEV_1045"',
+                    '"ACPI\\QEMU0001"', '"PCI\\VEN_1AF4&DEV_1052"',
+                    '"PCI\\VEN_1AF4&DEV_105A"',
+                    '"PCI\\VEN_1AF4&DEV_1001" "PCI\\VEN_1AF4&DEV_1042"',
+                    '"PCI\\VEN_1AF4&DEV_1004" "PCI\\VEN_1AF4&DEV_1048"',
+                    '"ACPI\\QEMU0002"']
+
+device_name_list = ["Red Hat VirtIO Ethernet Adapter", "VirtIO RNG Device",
+                    "VirtIO Serial Driver", "VirtIO Balloon Driver",
+                    "QEMU PVPanic Device", "VirtIO Input Driver",
+                    "VirtIO FS Device", "Red Hat VirtIO SCSI controller",
+                    "Red Hat VirtIO SCSI pass-through controller",
+                    "QEMU FWCfg Device"]
+
+
+def install_gagent(session, test, qemu_ga_pkg, gagent_install_cmd):
+    """Install guest agent."""
+    logging.info("Install 'qemu-guest-agent' package in guest.")
+    vol_virtio_key = "VolumeName like '%virtio-win%'"
+    vol_virtio = utils_misc.get_win_disk_vol(session, vol_virtio_key)
+    qemu_ga_pkg_path = r"%s:\%s\%s" % (vol_virtio, "guest-agent", qemu_ga_pkg)
+    gagent_install_cmd = gagent_install_cmd % qemu_ga_pkg_path
+    s_inst, o_inst = session.cmd_status_output(gagent_install_cmd)
+    if s_inst != 0:
+        test.fail("qemu-guest-agent install failed,"
+                  " the detailed info:\n%s." % o_inst)
+
+
+def uninstall_gagent(session, test, gagent_uninstall_cmd):
+    """Uninstall guest agent."""
+    logging.info("Try to uninstall 'qemu-guest-agent' package.")
+    s, o = session.cmd_status_output(gagent_uninstall_cmd)
+    if s:
+        test.fail("Could not uninstall qemu-guest-agent package ")
+
+
+def get_gagent_version(session, test, qemu_ga_pkg, gagent_install_cmd,
```

Comment:
The definition of this function seems confusing, I will recommend remove it and directly return the `gagent_version` in `install_gagent()`, it makes sense to provide a version after the installation of an application.